composer/loggers/slack_logger.py

```diff
@@ -1,64 +1,192 @@
+
+
+class SlackLogConfig:
+    """Log config for slack logger, including formatter function, keys to include, and interval in seconds.
+
+    formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+    include_keys (Set[str]): A set of metric/logs/traces keys to include in the message. If None, no keys are included.
+    interval_in_seconds (int): Interval in seconds to send logs to slack. If None, message is sent after every log call.
+    Note that no more than 50 items are allowed to send in a single message.
+    """
+
+    def __init__(
+        self,
+        formatter_func: Optional[Callable[..., List[Dict[str, Any]]]] = None,
+        include_keys: Optional[Set[str]] = None,
+        interval_in_seconds: Optional[int] = None,
+    ) -> None:
+        self.formatter_func = formatter_func
+        self.include_keys = include_keys
+        self.interval_in_seconds = interval_in_seconds
 
 
 class SlackLogger(LoggerDestination):
-    """Log metrics to slack, using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`.
+    """Logger to log metrics to slack.
 
-    .. note::
-        See example instantiating of SlackLogger in `examples/slack_log_metrics.py`
-        Webhook URL will be passed as an environment variable.
-        Formatter function can be defined at callsite or default formatter function will be applied if not provided.
+    This logger uses Slack's postMessage api - https://api.slack.com/methods/chat.postMessage.
 
     Args:
-        webhook_url (str): Slack Webhook URL
-        log_metrics_formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+        log_metrics_config (SlackLogConfig): Configuration for logging metrics to slack.
+        log_traces_config (SlackLogConfig): Configuration for logging traces to slack.
+        log_hparams_config (SlackLogConfig): Configuration for logging hyperparameters to slack.
     """
 
     def __init__(
         self,
-        webhook_url: str = '',
-        log_metrics_formatter_func: Optional[Callable] = None,
+        log_metrics_config: Optional[SlackLogConfig] = None,
+        log_traces_config: Optional[SlackLogConfig] = None,
+        log_hparams_config: Optional[SlackLogConfig] = None,
     ) -> None:
-        self.webhook_url = webhook_url
-        self.logged_metrics: Dict[str, float] = {}
-        self.log_metrics_formatter_func = log_metrics_formatter_func if log_metrics_formatter_func else self._default_log_bold_key_normal_value_pair_with_header
+        try:
+            import slack_sdk
+            del slack_sdk
+        except ImportError as e:
+            raise MissingConditionalImportError('health_checker', 'slack_sdk', None) from e
+
+        self.client = WebClient()
+        self.slack_logging_api_key = os.environ.get('SLACK_LOGGING_API_KEY', None)
+        self.channel_id = os.environ.get('SLACK_LOGGING_CHANNEL_ID', None)
```

Comment:
Changed this to use the post message api instead of webhooks, because this allows user to send messages to a channel directly.

Webhooks with custom channels has been deprecated, see https://glitch.com/~slack-channel-webhook-blueprint